### PR TITLE
[Merged by Bors] - TY-2457 provide excluded sources to provider

### PR DESCRIPTION
**Pull request by rustonaut**
**References:**

- [story TY-2528](https://xainag.atlassian.net/browse/TY-2528)
- [task TY-2457](https://xainag.atlassian.net/browse/TY-2457)

**Based on:**

- #258 
- #259

### CODE REVIEW

**Comment by rustonaut** [comment]
In this PR I treated it exactly like market, but we probably want to do some refactoring in a follow up PR to:

- reduce code duplication between the current Ops implementations
- clean up some code around the configuration/locks etc.

**Comment by acrrd** [comment]
If it's based on other pr you can set them as the base branch

File: discovery_engine_core/core/src/engine.rs
Comment (janpetschexain) [inline discovery_engine_core/core/src/engine.rs:143]: 🟢  the wrapped types all implement `Default`, so we could simplify the call
```suggestion
            excluded_sources: Arc::default(),
```

File: discovery_engine_core/core/src/stack/ops/breaking.rs
Comment (janpetschexain) [inline discovery_engine_core/core/src/stack/ops/breaking.rs:67]: 🔴  this won't fetch any new items in the case of `(Some(markets), None)`, which is a viable case we need to support. same for personalized news.

File: discovery_engine_core/core/src/stack/ops/breaking.rs
Comment (janpetschexain) [inline discovery_engine_core/core/src/stack/ops/breaking.rs:126]: 🟡  we could wrap the vector in another `Arc` to avoid cloning the vector everytime we make a request, ie `BreakingNews` would have the field `excluded_sources: Option<Arc<RwLock<Arc<Vec<String>>>>>`. similarly for `PersonalizedNews`, wdyt?
```suggestion
    excluded_sources: Arc<Vec<String>>,
```

**Review by janpetschexain — COMMENTED** [review]
i won't hit the request changes button to not block this pr next week, but the red comment needs to be adressed before merging.

File: discovery_engine_core/core/src/stack/ops/breaking.rs
Comment (rustonaut) [reply discovery_engine_core/core/src/stack/ops/breaking.rs:126]: Yes I wanted to fix the unnecessary cloning in a follow up refactoring (which also reduces the number of locks and arcs) through until then but this works, too.

**Comment by rustonaut** [comment]
After (in the background) going through the borse -> rebase  -> borse cycle  for the PRs this is based on this is now finally fully re-based on `main`. All needed changes are applied. @Robert-Steiner

**Review by Robert-Steiner — APPROVED** [review]

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**
